core/src/main/java/com/airbnb/aerosolve/core/util/Util.java

```diff
@@ -302,6 +302,19 @@ public static double logBase2(double value) {
     return flatFeature;
   }
 
+  public static Map<String, List<Double>> denseFeatureDropout(
+      FeatureVector featureVector,
+      double dropout) {
+    Map<String, List<Double>> denseFeatures = featureVector.getDenseFeatures();
+    if (denseFeatures == null) return Collections.EMPTY_MAP;
+    Map<String, List<Double>> out = new HashMap<>();
+    for (Map.Entry<String, List<Double>> feature : denseFeatures.entrySet()) {
+      if (Math.random() < dropout) continue;
+      out.put(feature.getKey(), feature.getValue());
```

Comment:
in this case you are dropping the whole feature family? I feel you should be dropping some elements in a particular feature family not the entire family